sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/DiagnosticDescriptorBuilder.cs

```diff
@@ -30,15 +30,16 @@ public static class DiagnosticDescriptorBuilder
         public static readonly string MainSourceScopeTag = "MainSourceScope";
         public static readonly string TestSourceScopeTag = "TestSourceScope";
 
-        public static DiagnosticDescriptor GetUtilityDescriptor(string diagnosticId, string title) =>
+        public static DiagnosticDescriptor GetUtilityDescriptor(string diagnosticId, string title,
+            string[] customTags = null) =>
             new DiagnosticDescriptor(
                 diagnosticId,
                 title,
                 string.Empty,
                 string.Empty,
                 DiagnosticSeverity.Warning,
                 isEnabledByDefault: true,
-                customTags: BuildUtilityCustomTags());
+                customTags: customTags ?? BuildUtilityCustomTags());

```

Comment:
Adding transparent ability to configure which source types should the analyzer run on.